src/plugins/intel_cpu/src/nodes/fullyconnected.cpp

```diff
@@ -606,25 +606,23 @@ void FullyConnected::needSplitMemoryForTensorParallel() {
         tp_cfg.cached_dst = split_horizontal(context->getEngine(), dst, -1, tp_cfg.w_rank, tp_cfg.w_size, false);
 
         if (auto it = memory.find(ARG_DST | ARG_ATTR_SCALES); it != memory.end()) {
-            memory[ARG_DST | ARG_ATTR_SCALES] =
-                split_horizontal(context->getEngine(), it->second, 0, tp_cfg.w_rank, tp_cfg.w_size);
+            it->second = split_horizontal(context->getEngine(), it->second, 0, tp_cfg.w_rank, tp_cfg.w_size);
         }
 
         if (auto it = memory.find(ARG_WEI | ARG_ATTR_SCALES); it != memory.end()) {
             auto scale_mem = std::const_pointer_cast<IMemory>(it->second);
-            memory[ARG_WEI | ARG_ATTR_SCALES] =
-                attrs.weightsNonTransposed
-                    ? split_vertical(context->getEngine(), scale_mem, 0, tp_cfg.w_rank, tp_cfg.w_size)
-                    : split_horizontal(context->getEngine(), scale_mem, 0, tp_cfg.w_rank, tp_cfg.w_size);
+            it->second = attrs.weightsNonTransposed
+                             ? split_vertical(context->getEngine(), scale_mem, 0, tp_cfg.w_rank, tp_cfg.w_size)
+                             : split_horizontal(context->getEngine(), scale_mem, 0, tp_cfg.w_rank, tp_cfg.w_size);
         }
 
         if (auto it = memory.find(ARG_WEI | ARG_ATTR_ZERO_POINTS); it != memory.end()) {
             auto zeropoint_mem = std::const_pointer_cast<IMemory>(it->second);
             auto element_num = zeropoint_mem->getSize() / zeropoint_mem->getPrecision().size();
             if (element_num == 1) {
-                tp_cfg.cached_zeropoint = zeropoint_mem;
+                it->second = zeropoint_mem;
```

Comment:
What is the point of this assignment? It looks that zeropoint_mem is `it->second` from it's initialization `zeropoint_mem = std::const_pointer_cast<IMemory>(it->second)`. Is this what we really want to do?

Reply:
Just didn't want to change the code much, just to apply the fix.
There is actually more that can be done. For example, we probably do not need these tp_cfg.cache* entries anymore. We already have 'memory' object which acts as a cache for the input / output memory

Reply:
Removed